File: CH2-modules/EX4-automobile-costs/source-code/automobileCosts.cpp
#include <iostream>
using namespace std;

void inputExpenses(int &month, double &loanPayment, double &insurance, double &gas, double &oil, double &tires, double &maintenance)
{
    cout << endl;
    cout << "Expenses month " << month << endl;
    cout << "Enter a Loan Payment : $";
    cin >> loanPayment;
    cout << "Enter a Insurance    : $";
    cin >> insurance;
    cout << "Enter a Gas          : $";
    cin >> gas;
    cout << "Enter a Oil          : $";
    cin >> oil;
    cout << "Enter a Tires        : $";
    cin >> tires;
    cout << "Enter a Maintenance  : $";
    cin >> maintenance;
}

void processExpenses(int &month, double &loanPayment, double &insurance, double &gas, double &oil, double &tires, double &maintenance, double &totalMonthly, double &totalAnnual)
{
    totalMonthly = loanPayment + insurance + gas + oil + tires + maintenance;
    totalAnnual += totalMonthly;
}

void outputMonthlyExpenses(int &month, double &loanPayment, double &insurance, double &gas, double &oil, double &tires, double &maintenance, double &totalMonthly)
{
    cout << "Total : $" << totalMonthly << endl;
    month += 1;
    totalMonthly = 0;
}

void outputAnnualExpenses(double &totalAnnual)
{
    cout << endl;
    cout << "Total Annual Expenses : $" << totalAnnual << endl;
}

int main()
{
    int month = 1;
    double loanPayment = 0;
    double insurance = 0;
    double gas = 0;
    double oil = 0;
    double tires = 0;
    double maintenance = 0;
    double totalMonthly = 0;
    double totalAnnual = 0;

    inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
    processExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly, totalAnnual);
    outputMonthlyExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly);

    inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
    processExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly, totalAnnual);
    outputMonthlyExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly);

    inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
    processExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly, totalAnnual);
    outputMonthlyExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly);

    inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
    processExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly, totalAnnual);
    outputMonthlyExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly);

    inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
    processExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly, totalAnnual);
    outputMonthlyExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly);

    inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
    processExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly, totalAnnual);
    outputMonthlyExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly);

    inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
    processExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly, totalAnnual);
    outputMonthlyExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly);

    inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
    processExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly, totalAnnual);
    outputMonthlyExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly);

    inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
    processExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly, totalAnnual);
    outputMonthlyExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly);

    inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
    processExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly, totalAnnual);
    outputMonthlyExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly);

    inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
    processExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly, totalAnnual);
    outputMonthlyExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly);

    inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
    processExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly, totalAnnual);
    outputMonthlyExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance, totalMonthly);

    outputAnnualExpenses(totalAnnual);
}

File: CH2-modules/EX4-automobile-costs/source-code/automobileCosts.js
const input = require("prompt-sync")();

function inputExpenses(
  month,
  loanPayment,
  insurance,
  gas,
  oil,
  tires,
  maintenance
) {
  console.log("Expenses month       :  " + month.value);
  loanPayment.value = Number(input("Enter a Loan Payment : $"));
  insurance.value = Number(input("Enter a Insurance    : $"));
  gas.value = Number(input("Enter a Gas          : $"));
  oil.value = Number(input("Enter a Oil          : $"));
  tires.value = Number(input("Enter a Tires        : $"));
  maintenance.value = Number(input("Enter a Maintenance  : $"));
}

function processExpenses(
  loanPayment,
  insurance,
  gas,
  oil,
  tires,
  maintenance,
  totalMonthly,
  totalAnnual
) {
  totalMonthly.value =
    loanPayment.value +
    insurance.value +
    gas.value +
    oil.value +
    tires.value +
    maintenance.value;
  totalAnnual.value = totalAnnual.value + totalMonthly.value;
}

function outputMonthlyExpenses(month, totalMonthly) {
  console.log("Total Monthly Expenses : $" + totalMonthly.value);
  month.value = month.value + 1;
  totalMonthly.value = 0;
}

function outputAnnualExpenses(totalAnnual) {
  console.log("Total Annual Expenses : $" + totalAnnual.value);
}

function main() {
  let month = {
    value: 0,
  };
  let loanPayment = {
    value: 0,
  };
  let insurance = {
    value: 0,
  };
  let gas = {
    value: 0,
  };
  let oil = {
    value: 0,
  };
  let tires = {
    value: 0,
  };
  let maintenance = {
    value: 0,
  };
  let totalMonthly = {
    value: 0,
  };
  let totalAnnual = {
    value: 0,
  };

  inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
  processExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly,
    totalAnnual
  );
  outputMonthlyExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly
  );

  inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
  processExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly,
    totalAnnual
  );
  outputMonthlyExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly
  );

  inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
  processExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly,
    totalAnnual
  );
  outputMonthlyExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly
  );

  inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
  processExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly,
    totalAnnual
  );
  outputMonthlyExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly
  );

  inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
  processExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly,
    totalAnnual
  );
  outputMonthlyExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly
  );

  inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
  processExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly,
    totalAnnual
  );
  outputMonthlyExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly
  );

  inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
  processExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly,
    totalAnnual
  );
  outputMonthlyExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly
  );

  inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
  processExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly,
    totalAnnual
  );
  outputMonthlyExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly
  );

  inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
  processExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly,
    totalAnnual
  );
  outputMonthlyExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly
  );

  inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
  processExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly,
    totalAnnual
  );
  outputMonthlyExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly
  );

  inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
  processExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly,
    totalAnnual
  );
  outputMonthlyExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly
  );

  inputExpenses(month, loanPayment, insurance, gas, oil, tires, maintenance);
  processExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly,
    totalAnnual
  );
  outputMonthlyExpenses(
    month,
    loanPayment,
    insurance,
    gas,
    oil,
    tires,
    maintenance,
    totalMonthly
  );

  outputAnnualExpenses(totalAnnual);
}

// Entry Point
main();
